src/flb_lib.c

```diff
@@ -633,6 +633,7 @@ static void flb_lib_worker(void *data)
     struct flb_config *config;
 
     config = ctx->config;
+    flb_context_set(ctx);
```

Comment:
:+1: I'd suffered for SEGV on arbitrary places to trigger flb_reload(). Good fixing.